pennylane/devices/qubit/simulate.py

```diff
@@ -237,7 +257,297 @@ def simulate(
     tensor([0.68117888, 0.        , 0.31882112, 0.        ], requires_grad=True))
 
     """
-    state, is_state_batched = get_final_state(circuit, debugger=debugger, interface=interface)
+    if circuit.shots and has_mid_circuit_measurements(circuit):
+        return simulate_native_mcm(
+            circuit, rng=rng, prng_key=prng_key, debugger=debugger, interface=interface
+        )
+    state, is_state_batched, _ = get_final_state(circuit, debugger=debugger, interface=interface)
     if state_cache is not None:
         state_cache[circuit.hash] = state
     return measure_final_state(circuit, state, is_state_batched, rng=rng, prng_key=prng_key)
+
+
+# pylint: disable=too-many-arguments
+def simulate_native_mcm(
+    circuit: qml.tape.QuantumScript,
+    rng=None,
+    prng_key=None,
+    debugger=None,
+    interface=None,
+) -> Result:
+    """Simulate a single quantum script with native mid-circuit measurements.
+
+    Args:
+        circuit (QuantumTape): The single circuit to simulate
+        rng (Union[None, int, array_like[int], SeedSequence, BitGenerator, Generator]): A
+            seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``.
+            If no value is provided, a default RNG will be used.
+        prng_key (Optional[jax.random.PRNGKey]): An optional ``jax.random.PRNGKey``. This is
+            the key to the JAX pseudo random number generator. If None, a random key will be
+            generated. Only for simulation using JAX.
+        debugger (_Debugger): The debugger to use
+        interface (str): The machine learning interface to create the initial state with
+        state_cache=None (Optional[dict]): A dictionary mapping the hash of a circuit to the pre-rotated state. Used to pass the state between forward passes and vjp calculations.
+
+    Returns:
+        tuple(TensorLike): The results of the simulation
+    """
+    if circuit.shots.has_partitioned_shots:
+        results = []
+        for s in circuit.shots:
+            aux_circuit = circuit.copy()
+            aux_circuit._shots = qml.measurements.Shots(s)
+            results.append(simulate(aux_circuit, rng, prng_key, debugger, interface))
```

Comment:
```suggestion
            results.append(simulate_native_mcm(aux_circuit, rng, prng_key, debugger, interface))
```

At this point I think we can say we have shots and mid circuit measurements, so we don't need to go all the way back up to `simulate`.